lib/common/net.go

```diff
@@ -7,15 +7,46 @@ import (
 	"net/url"
 	"strconv"
 	"strings"
+	"time"
 )
 
 var DefaultEndpoint int = 12345
 
 func CheckPortInUse(port int) error {
-	_, err := net.DialTimeout("tcp", net.JoinHostPort("", strconv.FormatInt(int64(port), 10)), 10)
+	if port < 1 {
+		return errors.New("0 port is not available")
+	}
+	_, err := net.DialTimeout(
+		"tcp",
+		net.JoinHostPort("", strconv.FormatInt(int64(port), 10)),
+		100*time.Millisecond,
+	)
 	return err
 }
 
+func GetFreePort(excludes ...int) (port int) {
```

Comment:
That bit bothers me a lot:
- It's test utility, not production code, and should be in a test file (`lib/test.go`)
- It checks if a port is available by doing a TCP request to it (!!!)
- It's called in a loop 

A much saner approach would be to bind the servers to port `0` and then get the actual port number for the client to use. Even better would be not to go through the network stack for unittests, but we're unfortunately quite far away from that.

Reply:
@Geod24 `lib/test.go` is not good place for these funcs, but there are some funcs, which are used in several packages. If you have good way for that, please let me know :)

I know using `0` port, we can replace it. Recently I think, we should test with `HTTP2Network` instead of `MemoryNetwork` because `MemoryNetwork` is used only for testing. Without binding 'real' port in `HTTP2Network`, we can consider the fake http.